src/main/java/com/openshift/jenkins/plugins/pipeline/Auth.java

```diff
@@ -148,9 +157,47 @@ public boolean allowCertificate(final X509Certificate[] certificateChain) {
 
 	@Override
 	public boolean allowHostname(String hostname, SSLSession sslSession) {
```

Comment:
It is an implementation of https://docs.oracle.com/javase/7/docs/api/javax/net/ssl/HostnameVerifier.html

A potentially interesting point on *when* this is called.  For ease, I'll paste a comment / piece of analysis from the referenced issue driving this:

Per the javadoc at https://docs.oracle.com/javase/7/docs/api/javax/net/ssl/HostnameVerifier.html, there are no default implementations in the JDK.  I also see similar discussions in http://stackoverflow.com/questions/6031258/java-ssl-how-to-disable-hostname-verification .. ("There is no hostname verification in standard Java SSL sockets or indeed SSL, so that's why you can't set it at that level. Hostname verification is part of HTTPS (RFC 2818): that's why it manifests itself as javax.net.ssl.HostnameVerifier, which is applied to an HttpsURLConnection.")

There seems to be an inconsistency between what the `HostnameVerifier` javadoc says (dating back to java 1.4), spefically:

`During handshaking, if the URL's hostname and the server's identification hostname mismatch, the verification mechanism can call back to implementers of this interface to determine if this connection should be allowed. `

But what I see in the lower level http libraries that openshift-restclient and fabric8 use (okhttp3) does not seem to adhere to that.  If you look at https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java#L266-L277, it calls verify irregardless of the outcome of the handshake.

So it would seem this will get called on every connection.